src/screens/ProfileScreen/elements/ProfilePicture.js

```diff
@@ -24,36 +27,55 @@ const ProfilePicture = ({
   return (
     <View style={styles.wrapImageProfile}>
       <TouchableNativeFeedback onPress={onImageContainerClick}>
-        <CircleGradient fill={karmaScore} size={normalize(100)} width={normalize(6)}>
-          <View style={styles.profileImageContainer}>
+        {withKarma ? (
+          <CircleGradient
+            fill={karmaScore}
+            size={normalize(size)}
+            width={normalize(width)}
+            testId="images">
+            <View style={styles.profileImageContainer(size)}>
+              <Image
+                testId="images"
+                style={styles.profileImage(size, width)}
+                source={{
+                  uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
+                }}
+                resizeMode={FastImage.resizeMode.stretch}
+              />
+              {renderAddIcon()}
+            </View>
+          </CircleGradient>
+        ) : (
+          <View style={styles.profileImageContainer(size)} testId="images">
             <Image
-              style={styles.profileImage}
+              testId="images"
+              style={styles.profileImage(size, width)}
               source={{
                 uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
               }}
               resizeMode={FastImage.resizeMode.stretch}
             />
             {renderAddIcon()}
           </View>
-        </CircleGradient>
+        )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code is mostly fine, but there's a bit of redundancy in the rendering of the `Image` and `renderAddIcon()` components. These are rendered in both branches of the conditional statement, which can lead to code duplication and increased maintenance effort. Consider extracting this into a separate function.

```diff
-          {withKarma ? (
-            <CircleGradient
-              fill={karmaScore}
-              size={normalize(size)}
-              width={normalize(width)}
-              testId="images">
-              <View style={styles.profileImageContainer(size)}>
-                <Image
-                  testId="images"
-                  style={styles.profileImage(size, width)}
-                  source={{
-                    uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
-                  }}
-                  resizeMode={FastImage.resizeMode.stretch}
-                />
-                {renderAddIcon()}
-              </View>
-            </CircleGradient>
-          ) : (
-            <View style={styles.profileImageContainer(size)} testId="images">
-               <Image
-                testId="images"
-                style={styles.profileImage(size, width)}
-                 source={{
-                   uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
-                 }}
-                 resizeMode={FastImage.resizeMode.stretch}
-               />
-               {renderAddIcon()}
-             </View>
-          )}

+          {withKarma ? (
+            <CircleGradient
+              fill={karmaScore}
+              size={normalize(size)}
+              width={normalize(width)}
+              testId="images">
+              {renderProfileContent()}
+            </CircleGradient>
+          ) : renderProfileContent()}

// Add this new function
const renderProfileContent = () => (
  <View style={styles.profileImageContainer(size)} testId="images">
    <Image
      testId="images"
      style={styles.profileImage(size, width)}
      source={{
        uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
      }}
      resizeMode={FastImage.resizeMode.stretch}
    />
    {renderAddIcon()}
  </View>
);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->